src/components/UserRegistration.js

```diff
@@ -0,0 +1,450 @@
+import { useState, useEffect } from "react";
+import { Form, Button, Alert } from 'react-bootstrap';
+import { useNavigate } from 'react-router-dom';
+import { createNewUser, addApiRole } from "../api/index";
+import { userRegistrationData } from "../data/userRegistrationData";
+import { validateUserRegistration } from "../validations/validateUserRegistration";
+import { login } from "../api/login";
+import "../CSS/UserRegistration.css";
+
+const UserRegistration = () => {
+    let navigate = useNavigate(); 
+    const [step, setStep] = useState(1);
+    const [validated, setValidated] = useState(false);
+    const [alert, setAlert] = useState(false);
+    const [values, setValues] = useState(userRegistrationData);
+
+    useEffect(() => {
+        login({username:'admin@demo.io', password:'Demo123!'});
+    },[]);
+
+    const handleChange = (ev) => {
+        const { name, value } = ev.target
+        setValues({ ...values, [name]: value });
+    };
+
+    const handleSubmit = (ev) => {
+        ev.preventDefault();
+
+        const form = ev.currentTarget;
+        if (form.checkValidity() === false) {
+            ev.stopPropagation();
+        } 
+        setValidated(true);
+
+        if(!validateUserRegistration(values)){
+            setAlert(true);
+        } else {
+            const { title, firstName, lastName, gender, dob, ssn, emailAddress, password, address, country, locality, region, homePhone, workPhone, mobilePhone, postalCode } = values;
+            const newUser = {
+                title: title,
+                firstName: firstName,
+                lastName: lastName,
+                gender: gender,
+                dob: dob,
+                ssn: ssn,
+                emailAddress: emailAddress,
+                password: password,
+                address: address,
+                country: country,
+                locality: locality,
+                region: region,
+                homePhone: homePhone,
+                workPhone: workPhone,
+                mobilePhone: mobilePhone,
+                postalCode: postalCode,
+            };
```

Comment:
These lines aren't necessary. You can just pass the `values` object as a parameter to the `createNewUser` function.